be/src/http/action/mini_load.cpp

```diff
@@ -516,4 +622,230 @@ bool LoadHandleCmp::operator() (const LoadHandle& lhs, const LoadHandle& rhs) co
+                << ctx->brief();
+        return plan_status;
+    }
+    VLOG(3) << "params is " << apache::thrift::ThriftDebugString(ctx->put_result.params);
+    return Status::OK;
+}
+
+// new on_header of mini load
+Status MiniLoadAction::_on_new_header(HttpRequest* req) {
+    size_t body_bytes = 0;
+    size_t max_body_bytes = config::mini_load_max_mb * 1024 * 1024;
+    if (!req->header(HttpHeaders::CONTENT_LENGTH).empty()) {
+        body_bytes = std::stol(req->header(HttpHeaders::CONTENT_LENGTH));
+        if (body_bytes > max_body_bytes) {
+            std::stringstream ss;
+            ss << "file size exceed max body size, max_body_bytes=" << max_body_bytes;
+            return Status(ss.str());
+        }
+    } else {
+        evhttp_connection_set_max_body_size(
+                evhttp_request_get_connection(req->get_evhttp_request()),
+                max_body_bytes);
+    }
+
+    RETURN_IF_ERROR(check_request(req));
+
+    StreamLoadContext* ctx = new StreamLoadContext(_exec_env);
+    ctx->ref();
+    req->set_handler_ctx(ctx);
+
+    // auth information
+    if (!parse_basic_auth(*req, &ctx->auth)) {
+        LOG(WARNING) << "parse basic authorization failed." << ctx->brief();
+        return Status("no valid Basic authorization");
+    }
+
+    ctx->load_type = TLoadType::MINI_LOAD;
+    ctx->load_src_type = TLoadSourceType::RAW;
+
+    ctx->db = req->param(DB_KEY);
+    ctx->table = req->param(TABLE_KEY);
+    ctx->label = req->param(LABEL_KEY);
+    if(!req->param(SUB_LABEL_KEY).empty()) {
+        ctx->sub_label = req->param(SUB_LABEL_KEY);
+    }
+    ctx->format = TFileFormatType::FORMAT_CSV_PLAIN;
+    std::map<std::string, std::string> params(
+            req->query_params().begin(), req->query_params().end());
+    std::map<std::string, std::string>::iterator max_filter_ratio_it = params.find(MAX_FILTER_RATIO_KEY);
+    if (max_filter_ratio_it != params.end()) {
+        ctx->max_filter_ratio = strtod(max_filter_ratio_it->second.c_str(), nullptr);
+    }
+    std::map<std::string, std::string>::iterator timeout_it = params.find(TIMEOUT_KEY);
+    if (timeout_it != params.end()) {
+        ctx->timeout_second = std::stoi(timeout_it->second);
+    }
+    
+    LOG(INFO) << "new income mini load request." << ctx->brief()
+              << ", db: " << ctx->db << ", tbl: " << ctx->table;
+
+    // record metadata in frontend
+    RETURN_IF_ERROR(_begin_mini_load(ctx));
+
+    // open sink 
+    auto pipe = std::make_shared<StreamLoadPipe>();
+    RETURN_IF_ERROR(_exec_env->load_stream_mgr()->put(ctx->id, pipe));
+    ctx->body_sink = pipe;
+
+    // get plan from fe
+    RETURN_IF_ERROR(_process_put(req, ctx));
+
+    // execute plan
+    return _exec_env->stream_load_executor()->execute_plan_fragment(ctx);
+}
+
+void MiniLoadAction::_new_handle(HttpRequest* req) {
+    StreamLoadContext* ctx = (StreamLoadContext*) req->handler_ctx();
+    if (ctx == nullptr) {
```

Comment:
Is it possible that we get a null StreamLoadContext here?
I think here should be an assertion(DCHECK(ctx != nullptr)), not an if condition.

Reply:
Changed